object_templates/provider_type.erb

```diff
@@ -3,8 +3,16 @@ require 'puppet/resource_api'
 Puppet::ResourceApi.register_type(
   name: '<%= name %>',
   docs: <<-EOS,
-      This type provides Puppet with the capabilities to manage ...
-    EOS
+@summary Fancy new type
+@example Fancy new example
+apt_key { '6F6B15509CF8E59E6E469F327F438280EF8D349F':
+ source => 'http://apt.puppetlabs.com/pubkey.gpg'
+}
+
+This type provides Puppet with the capabilities to manage ...
+**Autorequires**:
```

Comment:
I would be inclined to remove the "autorequires" section

Reply:
I've found the autorequires to be very helpful. We use it extensively for the sensu module.

https://github.com/sensu/sensu-puppet/blob/sensu2/REFERENCE.md

Reply:
In the same vein as above, the template should not document support for something it doesn't actually implement. As a compromise we can add a "Further reading for developers" section here, that points out important things that folks should consider next after `pdk new provider`.